EventSourcing.EF/RecordContext.cs

```diff
@@ -0,0 +1,222 @@
+using EventSourcing.Core;
+using Microsoft.EntityFrameworkCore;
+
+namespace EventSourcing.EF;
+
+/// <summary>
+/// <see cref="DbContext"/> for Finaps.EventSourcing <see cref="Record"/> Types
+/// </summary>
+public class RecordContext : DbContext
+{
+  internal const string PreviousIndex = "PreviousIndex";
+  internal const string ZeroIndex = "ZeroIndex";
+  internal const int MaxTypeLength = 256;
+  
+  /// <inheritdoc />
+  public RecordContext() {}
+
+  /// <inheritdoc />
+  public RecordContext(DbContextOptions options) : base(options) {}
+
+  /// <summary>
+  /// Find all <see cref="Record"/>s and add them to the <see cref="RecordContext"/>
+  /// </summary>
+  /// <param name="builder"><see cref="ModelBuilder"/></param>
+  protected override void OnModelCreating(ModelBuilder builder)
+  {
+    // TODO: Make this easier to manually configure
+    
+    // Get all Records in Assembly
+    var records = GetAssemblyTypes<Record>();
+
+    // Get all Events, grouped by Aggregate Type
+    var events = records
+      .Where(type => typeof(Event).IsAssignableFrom(type))
+      .GroupBy(GetAggregateType).Where(x => x.Key != null)
+      .ToDictionary(grouping => grouping.Key!, grouping => grouping.ToList());
+    
+    // Get all Snapshots, grouped by Aggregate Type
+    var snapshots = records
+      .Where(type => typeof(Snapshot).IsAssignableFrom(type))
+      .GroupBy(GetAggregateType).Where(x => x.Key != null)
+      .ToDictionary(grouping => grouping.Key!, grouping => grouping.ToList());
+
+    // Get all projections
+    var projections = records
+      .Where(type => typeof(Projection).IsAssignableFrom(type))
+      .ToList();
+
+    // For each Aggregate Type, create one table containing all Events and another containing all Snapshots
+    // See https://docs.microsoft.com/en-us/ef/core/modeling/inheritance#table-per-hierarchy-and-discriminator-configuration
+    foreach (var aggregateType in events.Keys)
+    {
```

Comment:
Maybe we can move the content of these 3 foreach loops to 3 private methods CreateEventTable etc so that one can see in one glance what happens here.